app-engine/java/ultimate/src/com/google/cloud/tools/intellij/appengine/java/ultimate/impl/AppEngineStandardUltimateWebIntegration.java

```diff
@@ -96,26 +107,38 @@ public VirtualFile suggestParentDirectoryForAppEngineWebXml(
     }
 
     for (WebRoot webRoot : webRoots) {
-      VirtualFile parent = webRoot.getFile();
-      if (parent != null) {
-        if (WEB_INF.equals(parent.getName())) {
-          return parent;
+      VirtualFile webRootDir = webRoot.getFile();
+      if (webRootDir != null) {
+        if (WEB_INF.equals(webRootDir.getName())) {
+          return webRootDir;
         }
 
-        VirtualFile child = parent.findChild(WEB_INF);
-        if (child != null) {
-          return child;
+        VirtualFile webInfDir = webRootDir.findChild(WEB_INF);
+        if (webInfDir != null) {
+          return webInfDir;
         }
       }
     }
 
     try {
-      VirtualFile webRootFile = webRoots.get(0).getFile();
-      if (webRootFile != null) {
-        return VfsUtil.createDirectoryIfMissing(webRootFile, WEB_INF);
+      Optional<VirtualFile> webRootDir =
+          webRoots.stream().map(WebRoot::getFile).filter(Objects::nonNull).findFirst();
+
+      if (webRootDir.isPresent()) {
+        return VfsUtil.createDirectoryIfMissing(webRootDir.get(), WEB_INF);
+      } else {
+        // There is at least one web-root path, but none that exist, select the first and create it
+        String presentableUrl = webRoots.get(0).getPresentableUrl();
+        if (presentableUrl != null) {
+          VirtualFile newWebRoot = VfsUtil.createDirectories(presentableUrl);
+
+          if (newWebRoot != null) {
+            return VfsUtil.createDirectoryIfMissing(newWebRoot, WEB_INF);
```

Comment:
identical line with 128, could we use the same path possibly?

Reply:
done